### PR TITLE
Add switch network component

### PR DESCRIPTION
**Pull request by saeta-eth**
This PR adds a SwitchNetwork component. It's a cleaner version of what we already have, but there was an issue when the user changed networks and the wallet was not connected and now it's working properly.

### CODE REVIEW

File: components/SwitchNetwork/SwitchNetwork.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import { nameToIdMapping, chainIdMapping } from 'constants/networks';
+
+const SwitchNetwork = ({ pool, poolNetwork, isReady }: any) => {
+	const { network, walletAddress, provider } = Connector.useContainer();
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+
+	const poolNetworkId = nameToIdMapping[poolNetwork ?? '']?.id ?? '';
+
+	useEffect(() => {
```
Comment (0xcdb) [inline components/SwitchNetwork/SwitchNetwork.tsx:19]: You can probably use `useMemo` here

File: components/SwitchNetwork/SwitchNetwork.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { ethers, BigNumber } from 'ethers';
```
Comment (0xcdb) [inline components/SwitchNetwork/SwitchNetwork.tsx:1]: The component is a modal so it should be named `SwitchNetworkModal`

File: components/SwitchNetwork/SwitchNetwork.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import { OPTIMISM_NETWORKS } from '@synthetixio/optimism-networks';
+
+import Connector from 'containers/Connector';
+
+import Button from 'components/Button';
+import BaseModal from 'components/BaseModal';
+
+import { nameToIdMapping, chainIdMapping } from 'constants/networks';
+
+const SwitchNetwork = ({ pool, poolNetwork, isReady }: any) => {
```
Comment (0xcdb) [inline components/SwitchNetwork/SwitchNetwork.tsx:13]: Can you please add the types?

File: pages/pools/[...poolData]/index.tsx [hunk 4]
```diff
@@ -43,40 +39,7 @@ const Pool: FC = () => {
...
-	}, [poolNetworkId, provider]);
-
-	useEffect(() => {
-		if (pool == null && !userClosedModal && (!poolQuery.isLoading || poolQuery.failureCount > 0)) {
-			setIsModalOpen(true);
-		} else {
-			setIsModalOpen(false);
-		}
-	}, [network.id, pool, poolNetworkId, poolQuery, userClosedModal]);
+	const isReady = !poolQuery.isLoading || poolQuery.failureCount > 0;
```
Comment (0xcdb) [inline pages/pools/[...poolData]/index.tsx:42]: `isReady` => `isPoolLoaded`

**Review by 0xcdb — CHANGES_REQUESTED** [review]

File: components/SwitchNetwork/SwitchNetwork.tsx (hunk 1 above)
Comment (saeta-eth) [reply components/SwitchNetwork/SwitchNetwork.tsx:19]: We need to set a new state, I think `useEffect` fits better for this.

File: pages/pools/[...poolData]/index.tsx (hunk 4 above)
Comment (saeta-eth) [reply pages/pools/[...poolData]/index.tsx:42]: Done

File: components/SwitchNetwork/SwitchNetwork.tsx (hunk 2 above)
Comment (saeta-eth) [reply components/SwitchNetwork/SwitchNetwork.tsx:1]: Done

**Pull request merged**